lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -0,0 +1,255 @@
+            end
+
+          case get_headway_mode(stop_ids, headway, alert, now) do
+            {:active, time_range, headsign} ->
+              %{type: :headway_section, pill: pill, time_range: time_range, headsign: headsign}
+
+            :inactive ->
+              %{type: :normal_section, rows: visible_departures}
+          end
+        end)
+
+      Enum.map(slot_ids, fn slot_id ->
+        %DeparturesWidget{
+          screen: config,
+          section_data: sections,
+          slot_names: [slot_id]
+        }
+      end)
+    end
+  end
+
+  defp get_sections_data(sections, fetch_section_departures_fn, fetch_alerts_fn) do
+    sections
+    |> Task.async_stream(fn %Section{
+                              query: %Query{
+                                params: %Params{stop_ids: stop_ids} = params
+                              },
+                              headway: headway
+                            } = section ->
+      section_departures =
+        case fetch_section_departures_fn.(section) do
+          {:ok, section_departures} -> section_departures
+          _ -> []
+        end
+
+      section_alert = get_section_alert(params, fetch_alerts_fn)
+
+      section_route = get_section_route_from_alert(stop_ids, section_alert)
+
+      %{
+        departures: section_departures,
+        pill: section_route,
+        alert: section_alert,
+        headway: headway,
+        stop_ids: stop_ids
+      }
+    end)
+    |> Enum.map(fn {:ok, data} -> data end)
+  end
+
+  # Alert will only have a value for sections that are configured for a station's ID
+  defp get_section_route_from_alert(["place-" <> _ = stop_id], %Alert{
+         informed_entities: informed_entities
+       }) do
+    informed_entities
+    |> Enum.find_value(fn
+      %{route: route, stop: ^stop_id} -> route
+      _ -> nil
+    end)
+    |> String.downcase()
+    |> String.to_atom()
+  end
+
+  defp get_section_route_from_alert(_, _), do: nil
+
+  defp get_section_alert(
+         %Params{
+           stop_ids: stop_ids,
+           route_ids: route_ids,
+           direction_id: direction_id
+         },
+         fetch_alerts_fn
+       ) do
+    alert_fetch_params = [
+      direction_id: direction_id,
+      route_ids: route_ids,
+      stop_ids: stop_ids
+    ]
+
+    alert_fetch_params
+    |> fetch_alerts_fn.()
+    |> Enum.filter(fn
+      # Show a headway message only during shuttles and suspensions at temporary terminals.
+      # https://www.notion.so/mbta-downtown-crossing/Departures-Widget-Specification-20da46cd70a44192a568e49ea47e09ac?pvs=4#e43086abaadd465ea8072502d6980d8d
+      %Alert{effect: effect} when effect in [:suspension, :shuttle] -> true
+      _ -> false
+    end)
+    |> List.first()
+  end
+
+  defp get_headway_mode(_, _, nil, _), do: :inactive
+
+  defp get_headway_mode(
+         stop_ids,
+         %Headway{headway_id: headway_id},
+         section_alert,
+         current_time
+       ) do
```

Comment:
If a screen is configured with secondary bus departures according to these params:
```
"params": {
    "direction_id": "both",
    "route_ids": [],
    "route_type": null,
    "stop_ids": ["150", "151", "15000"]
}
```
and there is a bus alert for one of these stops, this function still gets executed and breaks because there are multiple stop_ids. (Actually, more accurately, things break on line 228, where the function def requires only 1 array element.)

Reply:
Nice catch. That bus alert should never have been returned from the API. Looks like `route_types` was not added back to the fetch params after some changes. With that added, things are working again.